modules/text/src/text_detectorCNN.cpp

```diff
@@ -0,0 +1,458 @@
+
+
+
+#endif
+    }
+
+
+
+#ifdef HAVE_CAFFE
+    Ptr<caffe::Net<float> > net_;
+#endif
+    //Size inputGeometry_;
+    int minibatchSz_;//The existence of the assignment operator mandates this to be nonconst
+    //int outputSize_;
+public:
+    DeepCNNTextDetectorCaffeImpl(const DeepCNNTextDetectorCaffeImpl& dn):
+        minibatchSz_(dn.minibatchSz_){
+        outputGeometry_=dn.outputGeometry_;
+        inputGeometry_=dn.inputGeometry_;
+        //Implemented to supress Visual Studio warning "assignment operator could not be generated"
+#ifdef HAVE_CAFFE
+        this->net_=dn.net_;
+#endif
+    }
+    DeepCNNTextDetectorCaffeImpl& operator=(const DeepCNNTextDetectorCaffeImpl &dn)
+    {
+#ifdef HAVE_CAFFE
+        this->net_=dn.net_;
+#endif
+        this->setPreprocessor(dn.preprocessor_);
+        this->inputGeometry_=dn.inputGeometry_;
+        this->inputChannelCount_=dn.inputChannelCount_;
+        this->outputChannelCount_ = dn.outputChannelCount_;
+        // this->minibatchSz_=dn.minibatchSz_;
+        //this->outputGeometry_=dn.outputSize_;
+        this->preprocessor_=dn.preprocessor_;
+        this->outputGeometry_=dn.outputGeometry_;
+        return *this;
+        //Implemented to supress Visual Studio warning "assignment operator could not be generated"
+    }
+
+    DeepCNNTextDetectorCaffeImpl(String modelArchFilename, String modelWeightsFilename,Ptr<ImagePreprocessor> preprocessor, int maxMinibatchSz)
+        :minibatchSz_(maxMinibatchSz)
+    {
+
+        CV_Assert(this->minibatchSz_>0);
+        CV_Assert(fileExists(modelArchFilename));
+        CV_Assert(fileExists(modelWeightsFilename));
+        CV_Assert(!preprocessor.empty());
+        this->setPreprocessor(preprocessor);
+#ifdef HAVE_CAFFE
+        this->net_.reset(new caffe::Net<float>(modelArchFilename, caffe::TEST));
+        CV_Assert(net_->num_inputs()==1);
+        CV_Assert(net_->num_outputs()==1);
+        CV_Assert(this->net_->input_blobs()[0]->channels()==1
+                ||this->net_->input_blobs()[0]->channels()==3);
+        //        this->channelCount_=this->net_->input_blobs()[0]->channels();
+
+
+
+        this->net_->CopyTrainedLayersFrom(modelWeightsFilename);
+
+        caffe::Blob<float>* inputLayer = this->net_->input_blobs()[0];
+
+        this->inputGeometry_.height = inputLayer->height();
+        this->inputGeometry_.width = inputLayer->width();
+        this->inputChannelCount_ = inputLayer->channels();
+        //this->inputGeometry_.batchSize =1;
+
+        inputLayer->Reshape(this->minibatchSz_,this->inputChannelCount_,this->inputGeometry_.height, this->inputGeometry_.width);
+        net_->Reshape();
+        this->outputChannelCount_ = net_->output_blobs()[0]->channels();
+        //this->outputGeometry_.batchSize =1;
+        this->outputGeometry_.height =net_->output_blobs()[0]->height();
+        this->outputGeometry_.width = net_->output_blobs()[0]->width();
+
+
+
```

Comment:
Please remove unnecessary blank lines at that end of code blocks. There should be 1 blank line to separate blocks of code and 2 blank lines at the end of functions. It's a style issue but being consistent makes it easier to read.